### PR TITLE
Remove "ft-" prefix from checkpoint shards.

### LINKED ISSUE #2345: "ft-" prefix for finetuned checkpoints
I noticed that finetuned checkpoints get saved with an `ft-` prefix. Is this required to support PEFT methods or just convention? This makes it slightly inconvenient to use finetuned checkpoints in existing configs because you have to remember to update the `filename_format` as well as the `checkpoint_dir`.

I realize this is overly nit-picky but can have outsized effects since it's easy to forget and only find out after waiting 20-30 min to get scheduled 🙂 .

### PR DESCRIPTION
**Pull request by EugenHotaj**
Fixes #2345.

#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [x] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
* As titled

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [x] run unit tests via `pytest tests`
- [x] run recipe tests via `pytest tests -m integration_test`
- [x] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [x] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by felipemello1** [comment]
thanks @EugenHotaj , just confirming, were you able to run some dummy job to check if everything works fine?

**Review by felipemello1 — APPROVED** [review]
merging when ci is green
